tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+# This file is part of ts_FiberSpectrograph.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import unittest
+import shutil
+
+import asynctest
+
+from lsst.ts import salobj
+from lsst.ts.FiberSpectrograph import AvsSimulator
+from lsst.ts.FiberSpectrograph import FiberSpectrographCsc
+from lsst.ts.FiberSpectrograph import AvsReturnError, AvsReturnCode
+from lsst.ts.idl.enums.FiberSpectrograph import ExposureState
+
+STD_TIMEOUT = 2  # standard command timeout (sec)
+LONG_TIMEOUT = 20  # timeout for starting SAL components (sec)
+
+
+class Harness:
+    """An configurable async context manager for setting up and starting a CSC,
+    and any other pieces it needs to talk to."""
+    def __init__(self, initial_state, config_dir=None):
+        salobj.test_utils.set_random_lsst_dds_domain()
+        self.csc = FiberSpectrographCsc(
+            initial_state=initial_state,
+            initial_simulation_mode=0)
+        self.remote = salobj.Remote(domain=self.csc.domain, name="FiberSpectrograph", index=0)
+
+    async def __aenter__(self):
+        await self.csc.start_task
+        await self.remote.start_task
+        return self
+
+    async def __aexit__(self, *args):
+        await self.csc.close()
```

Comment:
I suggest `await self.remote.close()` before closing the CSC (not after, because closing the CSC closes the domain).

Reply:
Another one taken from ATDome. We should write that "how to test a CSC" doc!

Reply:
Good catch. I'll fix it in ts_ATDome.